### PR TITLE
adds a new command to set the date and time

### PR DESCRIPTION
**Pull request by tharts**
The oven's time runs ahead or past the actual time after a while and has to be set.
The pull request adds a new command Set_DateTime with which the date and time of the MCZ oven can be set.
The value to the command has to be given as string in the format
"ddmmYYYYHHmm", e.g. "171220201636" for the date 17/12/2020 04:36 pm.

### CODE REVIEW

File: commands.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 MCZ Maestro Command class
 These are the supported commands to be set via websocket
 '''
+from datetime import datetime, timedelta
```
Comment (tharts) [inline commands.py:6]: import timedelta isn't used anywhere and may be safely removed

File: commands.py [hunk 2]
```diff
@@ -36,6 +37,7 @@ def __init__(self, maestrocommand, commandvalue):
 MAESTRO_COMMANDS.append(MaestroCommand('Power', 34, 'onoff40'))
 MAESTRO_COMMANDS.append(MaestroCommand('Fan_State', 37, 'int'))# 0, 1, 2, 3 ,4,  5 ,6
 MAESTRO_COMMANDS.append(MaestroCommand('Control_Mode', 40, 'onoff')) # 0 = Auto , 1 = Manual
+MAESTRO_COMMANDS.append(MaestroCommand('Set_DateTime', 0, 'datetime')) # Format ddmmYYYYHHMM
```
Comment (tharts) [inline commands.py:40]: Not absolutely shure if the id should be 0 or 1. Below is the original message send to the stove from the app:
`42["chiedo",{"serialNumber":"1901290060006","macAddress":"86F3EB6DEF87","tipoChiamata":1,"richiesta":"C|SalvaDataOra|161220201128"}]`

File: maestro.py [hunk 3]
```diff
@@ -162,7 +168,7 @@ def process_info_message(message):
 
 def on_message(ws, message):
     message_array = message.split("|")
-    if message_array[0] == MaestroMessageType.Info.value:
+    if message_array[0] == MaestroMessageType.Info.value | message_array[0] == MaestroMessageType.String.value:
```
Comment (tharts) [inline maestro.py:171]: Not absolutely shure if it's the correct way to handle it. The stove sends back a message:
`42["rispondo",{"stringaRicevuta":"AA|171220202354","socketChiamata":"LvpI_guuJgmxzryC5qQr"}]`

Comment (Chibald) [reply maestro.py:171]: Dont think you can do anything with that. Here's the code that i found from the android app, it justs eats the stringrecvutta "AA" response.

```
function elaboraStringaRicevuta( stringaPassata )
{
    var datiRicevuti = stringaPassata.split("|");
    var end = new Date().getTime();
    var stringa = stringaPassata.substring(3);
    var dispositivoConnesso = $("#sub-lista-dispositivi .cambia-dispositivo[mac='" + dispositivoAssociato[1] + "'] .img-selected");
    var dispositivoConnesso2 = $(".popup-modifiche .dispositivi-associati .cambia-dispositivo[mac='" + dispositivoAssociato[1] + "'] .img-selected")
    
    if( datiRicevuti[0] == "PING" )
            {$( "#sub-info .recupero-stato-array .recupero-versione-pannello-direct .valore-recuperato" ).html( datiRicevuti[1] ); sendCommand( "PONG", "PINGPONG" );}
    else if( datiRicevuti[0] == "00" && stringa.length > 0 ) 
        {statoConnessione = 30; funzioniGenerali(stringa, "00"); nascondiDivAlert(); statoConnessioneOld = 20; $(".popup-modifiche .close-popup").trigger("click");}
    /*else if( datiRicevuti[0] == "00" ) // 
        {
            if( tipoConnessione == "direct" ) {statoConnessione = 100; console.log("00 corta:");}
            if( tipoConnessione == "remoto" ) {statoConnessione = 20; statoConnessioneTime = new Date().getTime();}
        }*/
    else if( datiRicevuti[0] == "01" && stringa.length >= 0 )
        {
         
            statoConnessione = 40; timeChiamate = new Date().getTime(); funzioniGenerali(stringa, "01"); 
            if( statoConnessioneOld == 20 )
                {
                    dispositivoConnesso.html('<img src="images/stove.png" width="100%" />');
                    dispositivoConnesso2.html('<img src="images/stove.png" width="100%" />');
                    if( sistemaOperativo == 0 ) {
                        navigator.notification.activityStop();
                    } 
                    statoConnessioneOld = 30;
                }
            //if( primaAperturaAPP == 0 ) {$("#sub-power").removeClass("hidden"); $("#sub-lista-dispositivi").addClass("hidden"); primaAperturaAPP = 1;}
        }
    /*else if( datiRicevuti[0] == "01" )  // 
        {
            if( tipoConnessione == "direct" ) {statoConnessione = 100; console.log("01 corta:");}
            if( tipoConnessione == "remoto" ) {statoConnessione = 30;}
        }*/
    else if( datiRicevuti[0] == "AA" )
        {}
    else if( stringa.length > 0 ) 
        {funzioniGenerali(stringa, datiRicevuti[0], datiRicevuti[1]); $("#xLoader").fadeOut("slow");}
    /*else if( stringa.length == 0 )
        {
            if( tipoConnessione == "direct" ) {statoConnessione = 100; console.log("Stringa Vuota:");}
            if( tipoConnessione == "remoto" ) {socket.emit('disconnect'); statoConnessioneTime = new Date().getTime() - 7000; statoConnessione = 20;}
        }*/
    
    console.log("Stato: " + statoConnessione + " - MS:", end - start, " - Stringa: " + stringa );
}

```

File: commands.py (hunk 2 above)
Comment (Chibald) [reply commands.py:40]: MAESTRO_COMMANDS  are meant to privide a list of WriteParametri command ID's. So, power on of will be C|WriteParametri|34|

The set datetime is a diiferent command by itself. It is 
C|SalvaDataOra|

0 or 1 will have no meaning there

**Comment by Chibald** [comment]
Iit's been 2 years, but I finally came around to make a new version of the script, and incorporated the "setdatetime" command .
becasuse the code changed a bit since the i copied it in anther PR. 

Thanks!

**Pull request closed without merging**